src/java.management.rmi/share/classes/javax/management/remote/rmi/RMIConnection.java

```diff
@@ -956,16 +956,17 @@ public void removeNotificationListener(ObjectName name,
      * @param filters an array of marshalled representations of the
      * <code>NotificationFilters</code>.  Elements of this array can
      * be null.
-     * @param delegationSubjects must be {@code null}.
+     * @param delegationSubjects should be {@code null}, but a non-null
+     * array is also accepted for compatibility reasons, which must not
+     * contain any non-null entries.
```

Comment:
The wording is bit unusual for a parameter description. Just wondering if might be clearer to say "null or an array of null elements" and put add an `@apiNote` to explain that it allows an array with null elements for compatibility reasons. What you have is okay too course, I'm just trying to think of another way to present this odd case.

Reply:
How about "must be null or an array of all null entries". You could still have an @apiNote explaining why.

Reply:
Thanks, appreciate the effort trying to make it perfect.  
Can't quite say "must be null or an array of all null entries" ..because I suppose it could be an empty array.

In reality, the only caller is our code that wraps a null Subject value, in an array, so it's generally a single null in an array.  

I hope we are OK sticking with "which must not contain any non-null entries" as that does cover it (and implicitly does tell you an empty array is fine).